frontends/p4/strengthReduction.cpp

```diff
@@ -288,6 +288,12 @@ const IR::Node* DoStrengthReduction::postorder(IR::Slice* expr) {
             return new IR::Concat(expr->type, expr,
                     new IR::Constant(IR::Type_Bits::get(-(lo + shift_amt)), 0)); } }
 
+    while (auto cast = expr->e0->to<IR::Cast>()) {
+        if (size_t(expr->getH()) < cast->expr->type->width_bits()) {
```

Comment:
Why does this work for a signed cast that widens the number?

Reply:
This dealing with the case where the slice make the cast irrelevant -- eg, if its a widening cast, but the slice only gets bits from the original (pre-cast) value, the extra widened bits are just thrown away anayways.  So the cast is irrelvant and can be removed.  The test here is 'are all the bits used by the slice present before the cast?'  If so drop the cast.